TabloidMVC/Controllers/PostController.cs

```diff
@@ -73,5 +92,82 @@ private int GetCurrentUserProfileId()
             string id = User.FindFirstValue(ClaimTypes.NameIdentifier);
             return int.Parse(id);
         }
+
+        
+        // GET: PostsController/Edit
+        public ActionResult Edit(int id)
+        {
+            List<Category> categories = _categoryRepository.GetAll();
+            Post post = _postRepository.GetPublishedPostById(id);
+
+            var vm = new PostCreateViewModel()
+            {
+                Post = post,
+                CategoryOptions = categories
+
+            };
+            vm.CategoryOptions = _categoryRepository.GetAll();
+
+
+            if (post == null)
+            {
+                return NotFound();
+            }
+
+            return View(vm);
+        }
+
+        // POST: PostsController/Edit
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public ActionResult Edit(int id, Post post)
+        {
+            List<Category> categories = _categoryRepository.GetAll();
+            
+            var vm = new PostCreateViewModel()
+            {
+                Post = post,
+                CategoryOptions = categories
+
+            };
+            vm.CategoryOptions = _categoryRepository.GetAll();
+            try
+            {
+                vm.Post.IsApproved = true;
+                post.UserProfileId = GetCurrentUserProfileId();
+                _postRepository.UpdatePost(post);
+
+                return RedirectToAction("Index");
+            }
+            catch (Exception ex)
+            {
+                return View(post);
+            }
+        }
+
+        // GET: PostController/Delete
```

Comment:
Delete is a separate ticket. You may comment out the delete functionality and put it in a different pr. I'm sure it's already working.